packages/scm/src/browser/scm-service.ts

```diff
@@ -0,0 +1,325 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { Disposable, DisposableCollection, Emitter, Event } from '@theia/core/lib/common';
+import { injectable } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+
+export const ScmService = Symbol('ScmService');
+export interface ScmService extends Disposable {
+
+    readonly onDidAddRepository: Event<ScmRepository>;
+    readonly onDidRemoveRepository: Event<ScmRepository>;
+
+    readonly repositories: ScmRepository[];
+    selectedRepository: ScmRepository | undefined;
+    readonly onDidChangeSelectedRepositories: Event<ScmRepository>;
+
+    registerScmProvider(provider: ScmProvider): ScmRepository;
+}
+
+export interface ScmProvider extends Disposable {
+    readonly label: string;
+    readonly id: string;
+    readonly handle: number;
+    readonly contextValue: string;
+
+    readonly groups: ScmResourceGroup[];
+
+    readonly onDidChangeResources: Event<void>;
+
+    readonly rootUri?: string;
+    readonly count?: number;
+    readonly commitTemplate?: string;
+    readonly onDidChangeCommitTemplate?: Event<string>;
+    readonly onDidChangeStatusBarCommands?: Event<ScmCommand[]>;
+    readonly acceptInputCommand?: ScmCommand;
+    readonly statusBarCommands?: ScmCommand[];
+    readonly onDidChange: Event<void>;
+
+    getOriginalResource(uri: URI): Promise<URI | undefined>;
+}
+
+export interface ScmResourceGroup {
+    readonly handle: number,
+    readonly sourceControlHandle: number,
+    readonly resources: ScmResource[];
+    readonly provider: ScmProvider;
+    readonly label: string;
+    readonly id: string;
+    readonly hideWhenEmpty: boolean | undefined;
+    readonly onDidChange: Event<void>;
+}
+
+export interface ScmResource {
+    readonly handle: number;
+    readonly groupHandle: number;
+    readonly sourceControlHandle: number;
+    readonly group: ScmResourceGroup;
+    readonly sourceUri: URI;
+    readonly decorations?: ScmResourceDecorations;
+    readonly selected?: boolean;
+
+    open(): Promise<void>;
+}
+
+export interface ScmResourceDecorations {
+    icon?: string;
+    tooltip?: string;
+    strikeThrough?: boolean;
+    faded?: boolean;
+
+    source?: string;
+    letter?: string;
+    color?: string;
+}
+
+export interface ScmCommand {
```

Comment:
Why do we need special `ScmCommand` and cannot use core `Command`? What is the difference?

Reply:
`tooltip` is missed in `Command` from Core